### ISSUE TITLE
feat: Add event manipulation and customization features

### ISSUE THREAD

**Issue by abe-101** [mergecal/python-mergecal#21]
### Is your feature request related to a problem? Please describe.

no

### Describe alternatives you've considered

Currently the package focuses on merging calendars. I propose we consider also adding the abillity to manipulate/ customize event details.

Examples could include:

- Summary Prefix
- Summary Postfix
- Description Prefix
- Description Postfix
- Remove events based on keywords
- block out details of events (summary, description or location - perhaps more)

## Discussion Points

API Design: How should we structure the API to accommodate these new features?
CLI Tool Design: What command-line interface would be most intuitive for users?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Are you willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time, and I know how to start.
- [ ] Yes, I have the time, but I don't know how to start. I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by niccokunzmann** [comment]
> Summary Prefix
Summary Postfix
Description Prefix
Description Postfix

One way to approach this in a generalized way is this:

## Constraints:

In the end, each of the attributes must be converted to ICS.
However attributes like SUMMARY and DESCRIPTION can be text. They also have parameters that might get lost... E.g. You can have a description that is HTML and TEXT at the same time or (have not seen it but possibly a language dispatch) [OWC](https://github.com/niccokunzmann/open-web-calendar/blob/67991b6fadc3bf75bdfecfe9c4d3e6ea0e4d55bd/open_web_calendar/convert_to_dhtmlx.py#L138).

## General Approach

The event attributes can be put through a template (jinja):

```yaml
summary: "[{{calendar.get('name', calendar.get('x-wr-calname', calendar['index'])) }}] {{ event['SUMMARY'] }}"
description: "{{ event['description'] }} \r\n from: {{ calendar['url'] }}" 
```

changes these attributes

```ics
SUMMARY: [..calendar name..] ..original summary..
DESCRIPTION: ..original description..
  from: ..calendar url..
```

**Comment by abe-101** [comment]
I like the idea of using a template engine but (to state the obvious) creating good documentation will become even more important.

**Comment by abe-101** [comment]
regarding the cli tool

We can design it in a way where users can point to a yaml file with their choose of template

**Comment by niccokunzmann** [comment]
The basic way is to use local files. It is not a problem to add downloading them for convenience however trying to duplicate the range of functionality of curl is out of scope, I think.
I would put urls into different issues and focus on the minimal example to implement bulk edits here.

**Comment by abe-101** [comment]
i agree that we should keep the approach simple and use local files.

What I'm am responding to is the implementation detail:
>  1. loop though each calendar, extend the calendar with custom attributes:
-[ ] X-CALMERGE-URL

My question is how does python-mergecal know the origin url of the calendar?
and to that my suggestion was to allow the user to pass it to the tool along side the calendar file

or perhaps its on the user to first set that attribute? :shrug: 

but perhaps I miss understood what you meant with the extended attribute

**Comment by niccokunzmann** [comment]
Oh, that X-CALMERGE-URL was just a thought which is 'nice to have' but not 'required'. For me, it was a way to show how we can add variables and where and show the extend to which one can use this method. For the first implementation, that specific attribute is not needed.
